sys/xtimer/xtimer_core.c

```diff
@@ -0,0 +1,494 @@
+
+static inline int _is_set(xtimer_t *timer)
+{
+    return (timer->target || timer->long_target);
+}
+
+void xtimer_init(void)
+{
+    /* initialize low-level timer */
+    timer_init(XTIMER, 1 /* us_per_tick */, _periph_timer_callback);
+
+    /* register initial overflow tick */
+    _lltimer_set(0xFFFFFFFF);
+}
+
+static void _xtimer_now64(uint32_t *short_term, uint32_t *long_term)
+{
+    uint32_t before, after, long_value;
+
+    /* loop to cope with possible overflow of xtimer_now() */
+    do {
+        before = xtimer_now();
+        long_value = _long_cnt;
+        after = xtimer_now();
+
+    } while(before > after);
+
+    *short_term = after;
+    *long_term = long_value;
+}
+
+uint64_t xtimer_now64(void)
+{
+    uint32_t short_term, long_term;
+    _xtimer_now64(&short_term, &long_term);
+
+    return ((uint64_t)long_term<<32) + short_term;
+}
+
+void _xtimer_set64(xtimer_t *timer, uint32_t offset, uint32_t long_offset)
+{
+    DEBUG(" _xtimer_set64() offset=%" PRIu32 " long_offset=%" PRIu32 "\n", offset, long_offset);
+    if (!long_offset) {
+        /* timer fits into the short timer */
+        xtimer_set(timer, (uint32_t) offset);
+    }
+    else {
+        xtimer_remove(timer);
+
+        _xtimer_now64(&timer->target, &timer->long_target);
+        timer->target += offset;
+        timer->long_target += long_offset;
+        if (timer->target < offset) {
+            timer->long_target++;
+        }
+
+        int state = disableIRQ();
+        _add_timer_to_long_list(&long_list_head, timer);
+        restoreIRQ(state);
+        DEBUG("xtimer_set64(): added longterm timer (long_target=%" PRIu32 " target=%" PRIu32 ")\n",
+                timer->long_target, timer->target);
+    }
+}
+
+void xtimer_set(xtimer_t *timer, uint32_t offset)
+{
+    DEBUG("timer_set(): offset=%" PRIu32 " now=%" PRIu32 " (%" PRIu32 ")\n", offset, xtimer_now(), _xtimer_now());
+    if (!timer->callback) {
+        DEBUG("timer_set(): timer has no callback.\n");
+        return;
+    }
+
+    xtimer_remove(timer);
+    uint32_t target = xtimer_now() + offset;
+
+    if (offset < XTIMER_BACKOFF) {
+        /* spin until timer should be run */
+        xtimer_spin_until(target);
+
+        _shoot(timer);
+    } else {
+        _xtimer_set_absolute(timer, target);
+    }
+}
+
+static void _periph_timer_callback(int chan)
+{
+    (void)chan;
+    _timer_callback();
+}
+
+static void _shoot(xtimer_t *timer)
+{
+    timer->callback(timer->arg);
+}
+
+static inline void _lltimer_set(uint32_t target)
+{
+    DEBUG("__lltimer_set(): setting %" PRIu32 "\n", _mask(target));
+    timer_set_absolute(XTIMER, XTIMER_CHAN, _mask(target));
+}
+
+int _xtimer_set_absolute(xtimer_t *timer, uint32_t target)
+{
+    uint32_t now = xtimer_now();
+    int res = 0;
+
+    DEBUG("timer_set_absolute(): now=%" PRIu32 " target=%" PRIu32 "\n", now, target);
+
+    timer->next = NULL;
+    if (target >= now && target - XTIMER_BACKOFF < now) {
```

Comment:
add parentheses to the conditions for better readability.
`if ((target >= now) && ((target - XTIMER_BACKOFF) < now))`